infrastructure/el/src/main/java/tech/pegasys/teku/infrastructure/el/DefaultExecutionClientProvider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.el;
```

Comment:
This doesn't make sense to put in infrastructure as it's not a generic library but very ethereum specific.  So it probably needs to be under the `ethereum` folder.  And we probably don't want to abbreviate to just `el` as that's a bit ambiguous.  Maybe a full `executionengine`?

Reply:
We cannot have all this in `executionlayer`, `ethereum` and `pow` if it will be part of one of these modules without circular dependecies

Reply:
Something like `ethereum/executionengine` could work. We need separate module for it, the infrastructure subfolder is definitely not a requirement as it's dedicated for core protocol logic